ansible_base/resource_registry/models/resource.py

```diff
@@ -55,6 +58,9 @@ class Resource(models.Model):
     # human readable name for the resource
     name = models.CharField(max_length=512, null=True)
 
+    created = models.DateTimeField(auto_now_add=True)
+    modified = models.DateTimeField(auto_now=True)
```

Comment:
I added the fields only to safely delete local orphan resources and avoid deleting resources that are in transient state.

In fact only one field is needed,the `modified`, but I added both to match the existing fields on other models.

I can remove those 2 fields and instead of filtering in the query set do a check using the related model, will be less efficient but then will not require the additional fields.

Reply:
Let's talk that out.

```python
    return Resource.objects.filter(
        created__lt=timezone.now() - timedelta(seconds=retain_seconds),
        service_id=manifest_list[0].service_id,
        content_type__resource_type__name=resource_type_name,
    ).exclude(ansible_id__in=[item.ansible_id for item in manifest_list])
```

This returns items that are identified as coming from the service `manifest_list[0].service_id`, but were not in the `manifest_list`. So why? When is this happening? An object is deleted in the provider service, and normally, we would not expect the _periodic sync_ to be cleaning this up, normally we would expect the provider service to make a DELETE request to the `/resources/:id/` endpoint to delete the item in the other service as a part of the transaction. If that works, I'm skeptical that this condition will _ever_ be hit. If it doesn't work, then sure, this is last resort, I get it.

I'm not compelled to worried about efficiency of this corner case.

The efficiency I _am_ worried about is that a periodic command grabs `manifest_list`, which self-apparently from even the code above, must contain _all_ objects of the given type. If we could use a `modified` time stamp to cut down on the number of items the periodic sync must obtain from the provider, that would be very worthwhile.

Reply:
This whole PR exists only as a rescue, it is to cover the last resort corner case, ideally this command will run and everything will be always `NOOP` because the server already propagated changes to each service.

The only way to cut the size of items returned by the manifest is if we persist the timestamp of the last sync and then take items modified after that time, is it worth adding another model to persist that data?

Next week I will do the work to remove the new fields and then check using an if condition on the created_lt of the related model.

Reply:
https://github.com/ansible/django-ansible-base/pull/297/commits/e5ee6d3f590ed69ba21262946b59fb2db62814f4

Reply:
Thanks, recapping, this no longer introduces timestamp fields on the `Resource` model, and uses the content object timestamp for reaping purposes.

I will see about filing an issue to open up planning design for the bigger issue of tracking timestamps on that model for making decisions about objects from the resource_server.